### PR TITLE
Refactor SQLite storage

### PR DESCRIPTION
**Pull request by marinoska**
- Refactor SQLite storage in order to embrace the recent changes to IDB storage
- Create `cojson-storage` package to accommodate synchronization logic common to SQLite and IDB storage
- Refactor SQLite and IDB storage in order to move SyncManager to the newly created `cojson-storage` package
- Make IDBClient and SQLiteClient implement the shared interface DBClientInterface

### CODE REVIEW

File: packages/cojson-storage/package.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "cojson-storage",
+  "version": "0.8.35",
```
Comment (marinoska) [inline packages/cojson-storage/package.json:3]: Not sure what should I state here for a new package

Comment (gdorsi) [reply packages/cojson-storage/package.json:3]: You made the right call to set the same version as the other packages.

I think it makes easier to manage dependencies this way.

**Review by gdorsi — APPROVED** [review]
LGTM

Very clean!

**Review by aeplay — CHANGES_REQUESTED** [review]
Looks very good, thanks @marinoska !

~~Small request: can we call the shared package `cojson-storage-relational`?~~

**Comment by marinoska** [comment]
> Looks very good, thanks @marinoska !
> 
> Small request: can we call the shared package `cojson-storage-relational`?

No problem! But why this name?

**Review by aeplay — APPROVED** [review]

**Pull request merged**